animal-sniffer/src/main/java/org/codehaus/mojo/animal_sniffer/ClassFileVisitor.java

```diff
@@ -155,15 +151,23 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
                 return FileVisitResult.CONTINUE;
             }
 
+            @Override
+            public FileVisitResult postVisitDirectory(final Path dir, final IOException exc) throws IOException {
+                for (final Path file : files) {
```

Comment:
Before processing any of the files it would be good to check whether `exc != null` and rethrow the exception in that case.